### PR TITLE
[WFCORE-683]: Wrong exception handling in ModuleLoadingResourceDefinition and use of jboss-module instead of reflection to get resource paths 

### PR DESCRIPTION
**Pull request by yersan**
Jira issue https://issues.jboss.org/browse/WFCORE-683

### CODE REVIEW

**Comment by ctomc** [comment]
this is ok to test

**Comment by kabir** [comment]
@yersan We need a bit of paperwork first:

We require that all contributions be made under the terms of the MIT License, http://www.opensource.org/licenses/mit-license.php. Do you agree to the these terms?

**Comment by yersan** [comment]
Hello @kabir 

Yes, I agree with the terms of the MIT License.

File: server/src/main/java/org/jboss/as/server/controller/resources/ModuleInfoHandler.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
     }
 
 
-    private ModuleLoaderMXBean getMxBean(ModuleLoader loader) throws ReflectiveOperationException {
+    protected ModuleLoaderMXBean getMxBean(ModuleLoader loader) throws ReflectiveOperationException {
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/controller/resources/ModuleInfoHandler.java]: Can you make this package protected? Thanks.

**Comment by bstansberry** [comment]
Besides the minor comment above, looks good. Can you squash this down to 2 commits? The first one and then squash the rest into a single commit.

**Comment by yersan** [comment]
Yes, of course, I'm going to change them in just two commits

**Comment by yersan** [comment]
It seems that something was wrong running the tests after reduce my commits in just two. @bstansberry Although you pointed me to do a squash commit, I did a reset in my local branch and commit my last three commits in just one. I will check it because the same test doesn't pass in my local computer

**Pull request merged**